src/Carnac.Logic/MessageController.cs

```diff
@@ -60,49 +66,29 @@ public void Start()
                         .Switch()
                         .Select(_ => message)
                         .Take(1);
-                })
-                .Select(m => Tuple.Create(m, ActionType.FadeOut));
+                });
+            var fadeOutMessageSubscription = fadeOutMessageStream
+                .ObserveOn(concurrencyService.MainThreadScheduler)
+                .SubscribeOn(concurrencyService.MainThreadScheduler)
+                .Subscribe(m => m.IsDeleting = true);
```

Comment:
I can't remember what you mentioned today at work, were you going to look at error handling in a later change?